onnx/backend/test/case/node/groupnorm.py

```diff
@@ -0,0 +1,64 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+
+import onnx
+
+from ..base import Base
+from . import expect
+
+
+class GroupNormalization(Base):
+    @staticmethod
+    def export() -> None:
+        def _groupnorm_test_mode(x, num_groups, scale, bias, epsilon=1e-5):
+            # Assume channel is first dim
+            assert x.shape[1] % num_groups == 0
+            group_size = x.shape[1] // num_groups
+            # Reshape to [N, group_size, C/group_size, H, W, ...]
+            new_shape = [x.shape[0], num_groups, group_size] + list(x.shape[2:])
+            x_reshaped = x.reshape(new_shape)
+            axes = tuple(range(2, len(new_shape)))
+            mean = np.mean(x_reshaped, axis=axes, keepdims=True)
+            var = np.var(x_reshaped, axis=axes, keepdims=True)
+            dim_ones = (1,) * (len(new_shape) - 2)
+            scale = scale.reshape(-1, *dim_ones)
+            bias = bias.reshape(-1, *dim_ones)
+            res = scale * (x_reshaped - mean) / np.sqrt(var + epsilon) + bias
+            return res.reshape(x.shape)
+
+        x = np.random.randn(3, 4, 2, 2).astype(np.float32)
+        num_groups = 2
+        scale = np.random.randn(num_groups)
+        bias = np.random.randn(num_groups)
```

Comment:
May be add `.astype(np.float32)` to bias and scale as well? Not sure if that is the default, but I notice x has it above.